aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -208,6 +208,27 @@
    */
   public static final String CLIENT_FACTORY = "client.factory";
 
+  /**
+   * The type of {@link software.amazon.awssdk.http.SdkHttpClient} implementation used by {@link AwsClientFactory}
+   * If set, all AWS clients will use this specified HTTP client.
+   * If not set, CLIENT_HTTP_TYPE_DEFAULT will be used.
+   * For specific types supported, see CLIENT_HTTP_TYPE_* defined below.
+   */
+  public static final String CLIENT_HTTP_TYPE = "client.http.type";
```

Comment:
I am a bit hesitated about this config name (although I proposed myself...), maybe we can have something better, @xiaoxuandev @amogh-jahagirdar @rajarshisarkar @singhpk234 if there is any better suggestion let us know!

Reply:
What seems more natural to me is http.client.type?

Reply:
I think the issue I was having with the original name was that it's a "client type", but there are 2 "client"s in the config name. `http.client.type` sounds good to me, it also matches the AWS class path for these clients, for example `software.amazon.awssdk.http.apache`. We don't really need the top level `client.` config, because it is configuring one level deeper.

I also thought about the potential configs in the future we want to add, such as connection and socket timeout, common configs can be after the same prefix, such as `http.client.connection-timeout`, and client specific ones can be for example `http.apache.connection-acquisition-timeout`.

@xiaoxuandev what do you think?

Reply:
Are we going to have different sublevel other than client under http? If no, maybe we should use http-client.type? If yes, then http.client.type looks fine to me.

Reply:
yeah `http-client.type` sounds good, we can go with that, the `.client` part is optional for shared HTTP configs, so we can have:
- http-client.type
- http-client.connection-timeout
- http-client.apache.connection-acquisition-timeout

what do you think @amogh-jahagirdar

Reply:
I am good with http-client.*!

Reply:
I am good with `http-client.*` as well.